### PR TITLE
Add diagnose code for NPE thrown in ExceptionPlaceHolder

### PR DESCRIPTION
**Pull request by blindpirate**
See https://github.com/gradle/gradle-private/issues/1368

There was a weird NPE in https://builds.gradle.org/viewLog.html?buildId=14073871&buildTypeId=Gradle_Check_Quick_Java8_Oracle_Linux_modelGroovy . This PR adds some diagnose code to catch the potential cause.

### CODE REVIEW

File: subprojects/messaging/src/main/java/org/gradle/internal/serialize/ExceptionPlaceholder.java [hunk 1]
```diff
@@ -151,10 +151,23 @@ public Object transform(Object obj) {
...
+        }
         return placeholder;
     }
 
+    private void diagnoseNullPointerException(NullPointerException e) {
+        // https://github.com/gradle/gradle-private/issues/1368
+        StringBuilder sb = new StringBuilder();
+        for (StackTraceElement element : stackTrace) {
+            sb.append(element).append("\n");
+        }
```
Comment (marcphilipp) [inline subprojects/messaging/src/main/java/org/gradle/internal/serialize/ExceptionPlaceholder.java]: Have you considered using `Throwable.printStrackTrace(new PrintWriter(new StringWriter()))` instead?

Comment (blindpirate) [reply subprojects/messaging/src/main/java/org/gradle/internal/serialize/ExceptionPlaceholder.java]: I don't quite understand. The problem is that there's a `StackTraceElement[]` which might contains a `null` and I want to print this malicious `StackTraceElement[]`, how can `Throwable.printStrackTrace()` help?

**Comment by blindpirate** [comment]
This is temporary diagnose code. You're right, I'll make it into log.

**Comment by blindpirate** [comment]
@marcphilipp If you don't object, I'll merge this one.

**Pull request merged**